temporalio/lib/temporalio/internal/worker/multi_runner.rb

```diff
@@ -0,0 +1,161 @@
+# frozen_string_literal: true
+
+require 'singleton'
+require 'temporalio/internal/bridge/worker'
+
+module Temporalio
+  module Internal
+    module Worker
+      class MultiRunner
+        def initialize(workers:)
+          @workers = workers
+          @queue = Queue.new
+
+          @shutdown_initiated_mutex = Mutex.new
+          @shutdown_initiated = false
+
+          # Start pollers
+          Bridge::Worker.async_poll_all(workers.map(&:_bridge_worker), @queue)
+        end
+
+        def apply_thread_or_fiber_block(&)
```

Comment:
Please confirm: this will only be used for evaluation of the block provided to the `Worker.run` or `Worker.run_all`, right?